integrationtests/src/test/java/io/kroxylicious/proxy/filter/CreateTopicRejectFilter.java

```diff
@@ -26,6 +27,6 @@ public void onCreateTopicsRequest(RequestHeaderData header, CreateTopicsRequestD
             topics.add(result);
         });
         response.setTopics(topics);
-        context.forwardResponse(response);
+        context.forwardResponse(new ResponseHeaderData().setCorrelationId(header.correlationId()), response);
```

Comment:
Why doesn't this fail [the equality](https://github.com/kroxylicious/kroxylicious/pull/296/files#diff-cfe73a8cf03d3c7af49b1c5330edd30444b988f472f55bf7457689242c28f361R192-R194) check in `forwardResponse`

Reply:
Because this is testing the short-circuit path where it's forwarding a response from a Request filter, this is only case where we support passing new objects through currently.

Reply:
though I do wonder if we want an overloaded operator where you only have to pass the body because then the framework can take care of this boilerplate copying the correlationId work.

Reply:
Yes, that sounds sensible.